2020/02/24/symfony-quickstart.html.md

```diff
@@ -0,0 +1,429 @@
+$request->request->get('category', 'default value')
+```
+
+is getting a post value.
+
+```php
+$request->query->all()
+```
+
+retrieves all GET parameters
+
+```php
+$request->request->all()
+```
+
+retrieves all POST parameters.
+
+## Caching
+
+Symfony supports caching with the caching module, which, if switched on will cache configuration and routes. If controller actions or routes are changed, then it is advisable to clear the cache via
+
+```
+bin/console cache:clear && bin/console cache:warmup
+```
+
+Whether or not to switch on caching for your application at dev mode depends on the frequency of a need to cache cleanup. Read more [here].
+
+## Configuration
+
+Symfony configuration can be done via YAML, XML or PHP files. In the projects I have been working with, configuration was done in YAML files and they were stored in the config folder inside the root folder of the project.
+
+Inside the **src** folder we can find **Kernel.php** which glues together the project.
+
+## Twig
+
+Symfony is well-paired with Twig, a popular template engine. Why don’t we use PHP for this purpose, one might ask. The answer is that we can use PHP or Smarty or whatever we intend to use for view templating, nobody forces us to use Twig.
+
+Using template engines instead of allowing the developers to use PHP as a template engine has the drawback of limiting the options, however, the exact reason they are used, especially by large teams is the exact same thing that we can outline as its drawback. Yes, it limits the options of the developers, but this way developers will not find it so easy and convenient to temporarily (?) implement their business logic as part of the view.
+
+If my opinion matters, I’m okay with using PHP as a template engine as long as we do not mix up the layers of the application and, by convention the view will remain “only” the view. I’m also okay with using template engines, even though it is less convenient to use when we need to do some experimenting and writing some code for a very short time, after which it is reverted. In that kind of experimental work one feels that enforcing this separation is unnecessary bureaucracy, but the gain is that we will not have to deal with views where lots of things are implemented which should not be there. So, this is a harmless form of bureaucracy, if such a thing exists.
+
+## Summary
+
+Symfony is a Framework which provides us with a way to work in conformity with the MVC pattern. Models are Entity classes along with their Repository pair, representing database tables, while Doctrine watches over the whole process, unless one chooses not to use Doctrine. Views are templates that generate HTML, for which Twig is a convenient tool, but one can use Symfony without relying on the usage of Twig. Configuration can be written in YAML, XML or PHP. PHP dependencies are handled by Composer, client-side dependencies are handled by Yarn. Symfony has its own console that can be used to run different commands.
```

Comment:
Again, client-side dependencies may be nonexistent, handled by Yarn, or by other things. AFAIK Symfony doesn't have any stance on that, right?

Reply:
@jonjensen I don't know about an official stance Symfony would have on the client-side dependencies, but the official documentation describes their support and overall my experience with Symfony is that these specific tools are strongly preferred by the community. Just like Doctrine and Twig is optional, client-side usage of Yarn, for example is optional as well. I will do a change to make sure this cannot be misunderstood.